Conceptpower+Spring/src/main/java/edu/asu/conceptpower/root/DatabaseClient.java

```diff
@@ -296,6 +302,84 @@ public boolean match(ConceptList dict) {
      * lang.String)
      */
     @Override
+    public List<ConceptEntry> getAllEntriesFromList(String listname, int page, int pageSize, final String sortBy,
+            final int sortDirection) {
+        ConceptEntry entry = new ConceptEntry();
+        entry.setConceptList(listname);
+
+        Query wordNetQuery = wordnetCacheClient.query();
+        wordNetQuery.constrain(ConceptEntry.class);
+        wordNetQuery.descend("conceptList").constrain(listname);
+
+        Query dictQuery = dictionaryClient.query();
+        dictQuery.constrain(ConceptEntry.class);
+        dictQuery.descend("conceptList").constrain(listname);
+
+        try {
+            final Field sortField = ConceptEntry.class.getDeclaredField(sortBy);
+            sortField.setAccessible(true);
+
+            Comparator<ConceptEntry> conceptEntryComparator = new Comparator() {
+
+                @Override
+                public int compare(Object o1, Object o2) {
+                    Object o1FieldContent;
+                    Object o2FieldContent;
+                    try {
+                        if (sortDirection == IConceptDBManager.ASCENDING) {
+                            o1FieldContent = sortField.get(o1);
+                            o2FieldContent = sortField.get(o2);
+                        } else {
+                            o2FieldContent = sortField.get(o1);
+                            o1FieldContent = sortField.get(o2);
+                        }
+                    } catch (IllegalArgumentException | IllegalAccessException e) {
+                        logger.error("Error accessing field.", e);
+                        return 0;
+                    }
+
+                    if (sortBy.endsWith("Date")) {
```

Comment:
I don't think that the date fields end with 'Date' in Conceptpower? Also, since it's using change events, this is kind of obsolete I think.